addon/components/x-toggle/component.js

```diff
@@ -37,57 +70,29 @@ export default Ember.Component.extend({
     return this.get('elementId') + '-x-toggle';
   }),
 
-  wasToggled: on('init', observer('toggled', function () {
-    var toggled = this.get('toggled');
-    var offIndex = this.get('offLabel').indexOf(':');
-    var onIndex = this.get('onLabel').indexOf(':');
-    var offState = offIndex > -1 ? this.get('offLabel').substr(offIndex + 1) : false;
-    var onState = onIndex > -1 ? this.get('onLabel').substr(onIndex + 1) : true;
-
-    this.sendAction('toggle', toggled, this.get('name'));
-
-    if (toggled === false) {
-      this.set('value', offState);
-    } else {
-      this.set('value', onState);
+  actions: {
+    onClick(e) {
+      e.stopPropagation();
     }
-  })),
-
-  valueObserver: on('init', observer('value', function() {
-    var debounce = this.get('debounce');
-
-    if (!debounce) {
-      debounce = run.debounce(this, function () {
-        var value = this.get('value');
-        var offIndex = this.get('offLabel').indexOf(':');
-        var onIndex = this.get('onLabel').indexOf(':');
-        var offState = offIndex > -1 ? this.get('offLabel').substr(offIndex + 1) : false;
-        var onState = onIndex > -1 ? this.get('onLabel').substr(onIndex + 1) : true;
-
-        if (value === onState) {
-          this.set('toggled', true);
-        } else {
-          this.set('toggled', false);
-          this.set('value', offState);
-        }
-
-        this.set('debounce', null);
-      }, 500);
-
-      this.set('debounce', debounce);
-    }
-  })),
-
-  clearDebounce: on('willDestroyElement', function () {
-    var debounce = this.get('debounce');
-
-    if (debounce) {
-      run.cancel(debounce);
-      this.set('debounce', null);
+  },
+
+  /**
+   * Provide a DDAU "action" or "mut" response
+   * @param  {string } action The name of the exposed action property
+   * @param  {hash}    hash   A hash of attributes that are passed back to a "action"
+   * @param  {mixed}   value  A value that is passed to the "update" function (aka, mut helper) if available
+   * @return {boolean}        Pass back true if `mut` not used; if used then proxies mut's response back
+   */
+  ddau(action, hash, value) {
+    if (this.attrs[action] && this.attrs[action].update) {
+      this.attrs[action].update(value);
```

Comment:
I thought that had to do with get and set? Until angle brackets the values are proxied through to the root object but with a one-way binding it was my impression that you _should_ use attrs. How else can you support the `mut` Helper?